redash/query_runner/big_query.py

```diff
@@ -31,29 +31,27 @@
     'TIMESTAMP': TYPE_DATETIME,
 }
 
+def transform_cell(field_type, cell_value):
+    if field_type == 'INTEGER':
+        return int(cell_value)
+    elif field_type == 'FLOAT':
+        return float(cell_value)
+    elif field_type == 'BOOLEAN':
+        return cell_value.lower() == "true"
+    elif field_type == 'TIMESTAMP':
+        return datetime.datetime.fromtimestamp(float(cell_value))
 
 def transform_row(row, fields):
-    column_index = 0
     row_data = {}
 
-    for cell in row["f"]:
+    for column_index, cell in enumerate(row["f"]):
         field = fields[column_index]
-        cell_value = cell['v']
-
-        if cell_value is None:
-            pass
-        # Otherwise just cast the value
-        elif field['type'] == 'INTEGER':
-            cell_value = int(cell_value)
-        elif field['type'] == 'FLOAT':
-            cell_value = float(cell_value)
-        elif field['type'] == 'BOOLEAN':
-            cell_value = cell_value.lower() == "true"
-        elif field['type'] == 'TIMESTAMP':
-            cell_value = datetime.datetime.fromtimestamp(float(cell_value))
+        if field['mode'] == 'REPEATED':
+            cell_value = [transform_cell(field['type'], item['v']) for item in cell['v']]
+        else:
+            cell_value = transform_cell(field['type'], cell['v'])
```

Comment:
If I'm not mistaken, `transform_cell` will now return None for any cell that has a type different than INTEGER/FLOAT/BOOLEAN/TIMESTAMP... You need a default return in `transform_cell`.

Reply:
Also you need to handle the case of `cell_value is None` there (many of the conversion will raise an exception if they get a `None` value).